core/domain/feedback_services.py

```diff
@@ -368,7 +421,126 @@ def _get_thread_from_model(thread_model):
         thread_model.id, thread_model.exploration_id, thread_model.state_name,
         thread_model.original_author_id, thread_model.status,
         thread_model.subject, thread_model.summary, thread_model.has_suggestion,
-        thread_model.created_on, thread_model.last_updated)
+        thread_model.message_count, thread_model.created_on,
+        thread_model.last_updated)
+
+
+def get_thread_summaries(user_id, full_thread_ids):
+    """Returns a list of summaries corresponding to each of the threads given.
+
+    Args:
+        user_id: str. The id of the user.
+        full_thread_ids: str. The complete ids of the threads for which we have
+            to fetch the summaries.
+
+    Returns:
+        list(dict). A list of dictionaries containing the summaries of the
+            threads given to it. The format of the returned value:
+            [
+                {
+                    'status': Status of the thread,
+                    'original_author_id': The id of the original author of the
+                        thread,
+                    'last_updated': When was the thread last updated,
+                    'last_message_text': The text of the last message,
+                    'total_no_of_messages': The total number of messages in the
+                        thread,
+                    'last_message_read': Whether the last message is read by the
+                        user,
+                    'second_last_message_read': Whether the second last message
+                        is read by the user,
+                    'author_last_message': The name of the author of the last
+                        message,
+                    'author_second_last_message': The name of the author of the
+                        second last message,
+                    'exploration_title': The title of the exploration to which
+                        exploration belongs
+                }
+            ]
+    """
+    exploration_and_thread_ids = (
+        [thread_id.split('.') for thread_id in full_thread_ids])
```

Comment:
Declare a function in feedback_models.FeedbackThreadModel to do the conversion instead of doing it here -- it's a counterpart to generate_full_thread_id(). Add tests for it.